base/unicode/utf8.jl

```diff
@@ -312,6 +312,11 @@ function convert(::Type{UTF8String}, a::Vector{UInt8}, invalids_as::AbstractStri
 end
 convert(::Type{UTF8String}, s::AbstractString) = utf8(bytestring(s))
 
+function convert{T<:Union{UInt16, UInt32}}(::Type{UTF8String}, a::Vector{T}, len=length(a)-1)
+    encode_to_utf8(T, a, len)
```

Comment:
The reason I did the encode functions that way (and I might have been incorrect in how best to acheive it, that was the very first bit of Julia code I worked on that was just a couple line wrapper to C code), was that I wanted to have an API that would support directly converting strings that were pointed to, were in shared memory, etc.